### ISSUE TITLE
internal server error due to file name restrictions? (not allowed characters? file name length exceeded?)

### ISSUE THREAD

**Issue by abubelinha** [zenodo/zenodo#2329]
Browser & OS: [Chrome 101 on Windows 7](https://www.whatismybrowser.com/w/CQTMCQ8)

## Describe the bug
I suspect some of my test uploads on sandbox.zenodo.org got corrupted because of "odd filenames".
Maybe the presence of some characters (`:`, `\`, `#`, `@`) is not allowed inside filenames?

## Steps to Reproduce
1. Go to https://sandbox.zenodo.org/record/1062720
    See Internal server error: `Error identifier: bee6407310254ba3b850f7e4123bb3b2`
2. Go to https://sandbox.zenodo.org/record/1062568
    See Internal server error: `Error identifier: 7605181c937c4f5fae48b7ce41baca21`

**IMPORTANT:** 
The error only appears when I am logged. So I guess only the record creator can see it?
If I log out I can see the published record. But as it is closed access (for now), file names are not visible. 
So perhaps this (filenames not exposed) is the reason for no "internal server error" in this case?

## Additional context
My uploads contain several files which I want to be accesible and downloadable on its own.

So instead of uploading a zipped file, I am uploading each file from a Python script based on this library:
https://github.com/SiLeBAT/zenodo-python

It just works fine, and [this method](https://github.com/SiLeBAT/zenodo-python/blob/master/zenodolib.py#L141) lets me specify target file names:
`deposition_files_create(self, deposition_id, target_name, file_path)` 

As I am recursively uploading all files in my local folder, I could find the same file name in different subpaths.
To avoid that conflict, I am using the whole file path as target_name for the files in Zenodo backup. Example files:

- C:\myfolder\path1\index.htm
`target_name =  'C:\myfolder\path1\index.htm'`
- C:\myfolder\path2\index.htm
`target_name =  'C:\myfolder\path2\index.htm'`

If I don't take the full path as target name, then I would have two files with the same name `index.htm`
Also, path information is important for understanding context of each file.

**My guess is I should not use cetain characters** in target file names,  like '`:`' or '`\`' 
I tried using '`#`' and '`@`' as replacements, but I am still getting errors (so perhaps one or both of them are also conflictive).
But as I cannot access the deposition files (due to the error), I can't know for sure which file names (and which characters) are the reason for the internal server error.

Note: my python script just uploads the files and metadata, but it does not publish the record.
The internal server error message appears after I manually publish the record using the web interface.

On the other hand, I can create a new version of the upload which contains files with "normal names", and then publish it with no issues. That's why I think the problem is in filenames.

### So, to summarize:
- Could you confirm if odd characters in file names are the reason of the above mentioned errors?
**Which specific characters are not allowed?** (and which replacement you suggest, if any)

- Also, is there **any limitation on filename lengths**?  
(some of my paths are pretty long but I think that was not the reason of the problem)

Thanks a lot in advance for your help.
